<?xml version="1.0"?>
<launch>
  <param name="robot_description" textfile="$(find mastering_ros_robot_description_pkg)/urdf/pan_tilt.urdf" />
  
  <!-- Starting joint publisher node which will publish the joint values -->
  <node 
    pkg="urdf_tutorial" 
    type="urdf_tutorial_a"
    name="joint_publisher" 
    launch-prefix="xterm -e"
  /> 

  <!-- Starting robot state publish which will publish tf -->
  <node 
    name="robot_state_publisher" 
    pkg="robot_state_publisher" 
    type="robot_state_publisher" 
  />

  <node pkg="turtlesim" name="turtle_teleop_key" type="turtle_teleop_key">
    <remap from="turtle1/cmd_vel" to="teleop_values"/>
  </node>


  <!-- Launch visualization in rviz -->
  <node 
    name="rviz" 
    pkg="rviz" 
    type="rviz" 
    args="-d $(find mastering_ros_robot_description_pkg)/urdf.rviz" 
    required="true" 
  />
</launch>
